osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyMainCirclePiece.cs

```diff
@@ -149,15 +129,17 @@ protected override void LoadComplete()
             if (hasNumber)
                 indexInCurrentCombo.BindValueChanged(index => hitCircleText.Text = (index.NewValue + 1).ToString(), true);
 
-            drawableObject.ApplyCustomUpdateState += updateStateTransforms;
-            updateStateTransforms(drawableObject, drawableObject.State.Value);
+            if (drawableObject != null)
+                drawableObject.ApplyCustomUpdateState += updateStateTransforms;
+
+            updateStateTransforms(drawableObject, drawableObject?.State.Value ?? ArmedState.Idle);
         }
 
         private void updateStateTransforms(DrawableHitObject drawableHitObject, ArmedState state)
         {
             const double legacy_fade_duration = 240;
 
-            using (BeginAbsoluteSequence(drawableObject.HitStateUpdateTime))
+            using (BeginAbsoluteSequence(drawableObject?.HitStateUpdateTime ?? 0))
```

Comment:
What's the deal with the null check added here? Applying a transform with an absolute sequence from `0` sounds never correct. Should either not have a null check in the first place, or avoid applying any transform if null (abort whole method early).

Reply:
Right, thought this was a delayed sequence. Will just abort method and call it a day.